### ISSUE TITLE
Wrapping in JSONObject and JSONArray inconsistent

### ISSUE THREAD

**Issue by johnjaylward** [stleary/JSON-java#273]
I'm debating if we need the `wrap` method at all. In some cases (like object construction) `wrap` is called and will possibly leave the object untouched (Integer, BigDecimal, String, etc) or will wrap it into a JSONObject and sometimes converted to a string value (if it's package name starts with "java.", "javax.", or if the classloader is null).

This leads to inconsistencies in how not only data is placed into the JSON container types, but also in how they are written out.

I'm proposing that we remove the wrap method altogether and always insert values into the JSONObject and JSONArray as-is and handle conversion on the serialization output (`write`, `toString`, etc).

If we don't remove the wrap method, then I propose that we update all the `put` methods to wrap unknown objects (https://github.com/stleary/JSON-java/blob/master/JSONArray.java#L810 and https://github.com/stleary/JSON-java/blob/master/JSONObject.java#L1273 at least) to maintain consistency.

I think removing the `wrap` method is a better solution as the output already handles any conversions needed. This would make all serialization "lazy" instead of some lazy and some eager depending on what way values were added to an object or array.

**Comment by stleary** [comment]
Even if we don't use it, wrap() must be kept since it is in the public API.

**Comment by johnjaylward** [comment]
While looking at the Number change PR, I did realize that the writeValue and valueToString methods don't properly serialize random beans that were added via the put method. If we did remove the wrap method from internal calls, we'd need to write more thorough tests on the output to verify beans are output correctly instead of calling bean.toString().

A faster fix would probably be to add the wrap calls to the put methods, although I'm not sure that's the most elegant or even the proper solution.
